wcomponents-theme/src/test/spec/wc.file.validate.test.mjs

```diff
@@ -0,0 +1,97 @@
+import validate from "wc/file/validate.mjs";
+import {addFilesToInput, getInput} from "../helpers/specUtils.mjs";
+describe("wc/file/validate", () => {
+	let testHolder;
+
+	function runWithEmptyCallbacks(extraArgs, testId, valid) {
+		let emptyCallback, emptyErrback;
+
+		const element = getInput(testHolder, testId);
+		addFilesToInput(element, [{ value: "SixteenChars.gif", type: "image/gif" }]);
+
+		emptyCallback = jasmine.createSpy("emptyCallback");
+		emptyErrback = jasmine.createSpy("emptyErrback");
+		validate.check({
+			selector: element,
+			callback: emptyCallback,
+			errback: emptyErrback,
+			...extraArgs
+		});
+
+		if (valid) {
+			expect(emptyCallback).toHaveBeenCalledTimes(1);
+			expect(emptyErrback).not.toHaveBeenCalled();
+		} else {
+			expect(emptyCallback).not.toHaveBeenCalled();
+			expect(emptyErrback).toHaveBeenCalledTimes(1);
+		}
+	}
+
+	function runWithFilesArg(files, testId, valid) {
+		let emptyCallback, emptyErrback;
+
+		const element = getInput(testHolder, testId);
+
+		emptyCallback = jasmine.createSpy("emptyCallback");
+		emptyErrback = jasmine.createSpy("emptyErrback");
+		validate.check({
+			selector: element,
+			callback: emptyCallback,
+			errback: emptyErrback,
+			files: files
+		});
+
+		if (valid) {
+			expect(emptyCallback).toHaveBeenCalledTimes(1);
+			expect(emptyErrback).not.toHaveBeenCalled();
+		} else {
+			expect(emptyCallback).not.toHaveBeenCalled();
+			expect(emptyErrback).toHaveBeenCalledTimes(1);
+		}
+	}
+
+	beforeEach(() => {
+		testHolder = document.body.appendChild(document.createElement("div"));
```

Comment:
It's no big deal, but since you are creating a new `testHolder` for each test, and leaving the previous testHolder lying around, maybe you should have an `afterEach` that removes them too, something like this:

```js
afterEach(() => {
		testHolder.parentElement.removeChild(testHolder);
```